tests/base.py

```diff
@@ -225,16 +230,39 @@ def expected_schema_keys(self, stream):
 
         return props.keys()
 
-    def modify_expected_datatypes(self, expected_records):
+    def sort_records_recur(self, records):
+        for record in records:
+            self.sort_record_recur(record)
+
+    def sort_record_recur(self, record):
+        if isinstance(record, dict):
+            for key, value in record.items():
+                if type(value) == dict:
+                    self.sort_record_recur(value)
+                elif type(value) == list:
+                    for rec in value:
+                        self.sort_record_recur(rec)
+                    self.sort_array_type(record, key, value)
+            
+
+    def modify_expected_records(self, records):
+       for rec in records:
+           self.modify_expected_record(rec)
+
+           
+    def modify_expected_record(self, expected_record):
         """ Align expected data with how the tap _should_ emit them. """
-        if type(expected_records) == list: # Modify a list of records
-            for record in expected_records:
-                for key, value in record.items():
-                    self.align_date_type(record, key, value)
-            return
+        if isinstance(expected_record, dict):
+            for key, value in expected_record.items(): # Modify a single record
+                if type(value) == dict:
+                    self.modify_expected_record(value)
+                elif type(value) == list:
+                    for item in value:
+                        self.modify_expected_record(item)
+                    self.sort_array_type(expected_record, key, value)
+                else:
+                    self.align_date_type(expected_record, key, value)
 
-        for key, value in expected_records.items(): # Modify a single record
-            self.align_date_type(expected_records, key, value)
 
     def align_date_type(self, record, key, value):
         """datetime values must conform to ISO-8601 or they will be rejected by the gate"""
```

Comment:
Ooops we still have to do this 1st To Do.

Reply:
It can wait but I think there will be some datetime schema issues revealed once this is done.